src/Storages/ObjectStorage/DataLakes/Iceberg/IcebergMetadata.cpp

```diff
@@ -107,6 +111,54 @@ void waitUntilCompressionIsDone(
     }
 }
 
+void scheduleCompactionJob(
+    ObjectStoragePtr object_storage,
+    StorageObjectStorageConfigurationPtr configuration,
+    const std::optional<FormatSettings> & format_settings,
+    SharedHeader sample_block,
+    ContextPtr context)
+{
+    auto compaction_period = context->getSettingsRef()[Setting::iceberg_period_compaction];
+    std::this_thread::sleep_for(std::chrono::milliseconds(compaction_period));
+    /// Compaction thread poll responses for compaction tasks.
+    context->getIcebergCompactionThreadPool().scheduleOrThrow([=] {
+        auto log = getLogger("IcebergCompaction");
+        try
+        {
+            Iceberg::compactIcebergTable(
+                object_storage,
+                configuration,
+                format_settings,
+                sample_block,
+                context
+            );
+        }
+        catch (Exception & ex)
+        {
+            LOG_DEBUG(log, "Iceberg Compaction failed {}", ex.what());
+            Iceberg::unlockCompaction(object_storage, configuration);
+        }
+        catch (...)
+        {
+            LOG_DEBUG(log, "Iceberg Compaction failed with unknown error");
+            Iceberg::unlockCompaction(object_storage, configuration);
+        }
+    });
+
+    /// Compaction Scheduler thread poll responses for task like "sleep for N seconds and schedule compaction of table".
+    context->getIcebergSchedulerCompactionThreadPool().scheduleOrThrow([=] {
```

Comment:
Should we better use a common `BackgroundSchedulePool` and explicitly define reschedule period via its `scheduleAfter` methods? It is done for many other table engines which need a background job. See for example https://github.com/ClickHouse/ClickHouse/blob/e7961eaf88c54ee4a3ade1eaa1ec264c5788cf2f/src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp#L634-L635
Also does the current way with compaction threadpool deactivates the task when table is dropped/shutdown?